site/content/docs/11-deploying.md

```diff
@@ -61,4 +61,70 @@ When deploying to [Now][], you can pass the environment variable into Now itself
 now -e SAPPER_TIMESTAMP=$(date +%s%3N)
 ```
 
-[Now]: https://zeit.co/now
\ No newline at end of file
+[Now]: https://zeit.co/now
+
+
+## Deploying with Docker
+
+[Docker](https://docs.docker.com/install/) can be used to build an image and run it on your local system. Doing this locally allows testing the build process before building and deploying it on a cloud platform. The Docker image can be build by adding a `Dockerfile` into your project:
+
+```Dockerfile
+# Dockerfile
+FROM mhart/alpine-node:latest as build
```

Comment:
It may be worth swapping this to one of the official nodejs alpine images. The originals are based off the mhart image, as far as I'm aware, but I'd imagine the node org have a bit more bandwidth for support and maintenance.

I think we should also suggest an image that pins at least the node major, perhaps the minor. something like `node:12-alpine3.11`, `node:12.16-alpine3.11` or `node:lts-alpine`. 

Any of these would be relatively safe bets, I'd generally encourage against using `latest` tags for production.

Reply:
There is also an argument for a more comprehensive image for tests. Sapper does have a cypress config, it makes sense to use a cypress image in that context. Although this is probably out of scope for these docs. Likewise, I don't know if cypress will be included for much longer.

Reply:
The image seems to get somewhat larger, but its all peanuts if you compare it to the total mass in the observable universe ;)

```
mhart/alpine-node:slim

REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
sapper-image        latest              c34e24e26222        5 seconds ago       74.5MB

node:lts-alpine

REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
sapper-image        latest              b2917ead41b1        3 seconds ago       89.6MB
```

Have updated the base images...

Reply:
Oh right, slim is with no npm or yarn installed. I don't think node to an alpine-slim image, all the mhart images are built on alpine, so that is my bad. `mhart/alpine-node:slim` may well be the best image after all.

Reply:
I also may have confused slim and stretch...

Since node don't seem to have the best possible images anyway, I am now confused as to what the best approach is. Just pick the best is probably fine, official image for the full node install and mharts slim image for prod.

Reply:
`mhart/alpine-node:slim-12` is probably a decent version to pin.

Reply:
Changed back to `mhart/alpine-node:slim-12` for production and `mhart/alpine-node:12` for build. Image size seems to clock in at 47.1MB...